src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -46,29 +47,44 @@ public EventCounter(string name, EventSource eventSource)
 
             InitializeBuffer();
             _name = name;
-            EventCounterGroup.AddEventCounter(eventSource, this);
+            _group = EventCounterGroup.GetEventCounterGroup(eventSource);
+            _group.Add(this);
         }
 
         /// <summary>
-        /// Writes the metric.
+        /// Writes 'value' to the stream of values tracked by the counter.  This updates the sum and other statistics that will 
+        /// be logged on the next timer interval.  
         /// </summary>
         /// <param name="value">The value.</param>
         public void WriteMetric(float value)
         {
             Enqueue(value);
         }
 
+        /// <summary>
+        /// Removes the counter from set that the EventSource will report on.  
+        /// </summary>
+        public void Dispose()
+        {
+            _group.Remove(this);
+        }
+
+        public override string ToString()
+        { 
+            return "EventCounter '" + _name + "' Count " + _count + " Mean " + (((double)_sum) / _count).ToString("n3");
```

Comment:
As you say, the cast to double was intended to avoid the divide by zero iss.  

The ToString() was added just to enhance debugging.    I would rather remove it than try to localize it.  I would prefer to keep it, but if you tell me to I will simply remove it.